manifests/init.pp

```diff
@@ -82,16 +82,16 @@
   include ::foreman_proxy
   include ::foreman_proxy::plugin::pulp
 
-  $pulp = $::foreman_proxy::plugin::pulp::pulpnode_enabled
-  if $pulp {
+  $pulp_node = $::foreman_proxy::plugin::pulp::pulpnode_enabled
+  if $pulp_node {
     assert_type(String[1], $pulp_oauth_secret)
   }
 
   $foreman_proxy_fqdn = $::fqdn
-  $foreman_url = "https://${parent_fqdn}"
-  $reverse_proxy_real = $pulp or $reverse_proxy
+  $foreman_url = $::foreman_proxy::foreman_base_url
+  $setup_reverse_proxy = $pulp_node or $reverse_proxy
```

Comment:
Can you explain cases where the `or` will come into play? Where you'd have one without the other and vice versa?